app/controllers/user_training_controller.rb

```diff
@@ -7,19 +7,26 @@
 class UserTrainingController < ApplicationController
   def index
     config = Talentlms::Config.first
-    if config.nil? || !current_user.training_required? || current_user.training_completed?
+    lms = Talentlms::Facade.new
+
+    # Verifying with local data before hitting the API. This prevents unneeded API calls
+    # and ensures local data is updated when new trainings have been completed.
+    if config.nil? || !current_user.training_required? || (current_user.training_completed? && !lms.training_expired?(current_user, false))
```

Comment:
I wonder if we should encapsulate 
```
!current_user.training_required? || (current_user.training_completed? && !lms.training_expired?(current_user, false)
```
in `lms.training_required?(current_user)`  or maybe `current_user.training_required?(lms)`.  I think we're always going to check all three items together, so this (and similar checks could just be):

```suggestion
    if config.nil? || !current_user.training_required?(lms)
```

Reply:
I've added this as a method in the lms Facade class. I've kept the ability to specify if we want to hit the API as the original logic is using these checks to see if the local data needs to be updated.